### PR TITLE
Create models

### PR DESCRIPTION
**Pull request by n-chornyi**

### CODE REVIEW

File: db/migrate/20211207110813_create_reviews.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class CreateReviews < ActiveRecord::Migration[6.1]
+  def change
+    create_table :reviews do |t|
+      t.string :title
+      t.string :text
+      t.timestamp :date
```
[the remark this answers is not in the document]
Comment (n-chornyi) [reply db/migrate/20211207110813_create_reviews.rb:6]: +

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
+  has_many :shipping_addresses
+  has_many :credit_cards
+  has_many :reviews
+  has_many :orders
```
[the remark this answers is not in the document]
Comment (n-chornyi) [reply app/models/user.rb:10]: +

File: app/models/book.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class Book < ApplicationRecord
+  belongs_to :author
+  has_many :order_items
```
[the remark this answers is not in the document]
Comment (n-chornyi) [reply app/models/book.rb:3]: +

File: app/models/author.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class Author < ApplicationRecord
+  has_many :books
```
[the remark this answers is not in the document]
Comment (n-chornyi) [reply app/models/author.rb:2]: +

File: db/migrate/20211207105429_devise_create_users.rb [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+      t.string :encrypted_password, null: false, default: ""
+
+      ## Recoverable
+      t.string   :reset_password_token
+      t.datetime :reset_password_sent_at
+
+      ## Rememberable
+      t.datetime :remember_created_at
+
+      ## Trackable
```
[the remark this answers is not in the document]
Comment (n-chornyi) [reply db/migrate/20211207105429_devise_create_users.rb:17]: +

File: db/migrate/20211207111139_create_orders.rb [hunk 6]
```diff
@@ -0,0 +1,11 @@
+class CreateOrders < ActiveRecord::Migration[6.1]
+  def change
+    create_table :orders do |t|
+      t.string :code_discount
+      t.boolean :status
```
[the remark this answers is not in the document]
Comment (n-chornyi) [reply db/migrate/20211207111139_create_orders.rb:5]: +

File: spec/factories/authors.rb [hunk 7]
```diff
@@ -0,0 +1,7 @@
+FactoryBot.define do
+  factory :author do
+    id { 1 }
+    first_name { 'John Ronald Reuel' }
```
[the remark this answers is not in the document]
Comment (n-chornyi) [reply spec/factories/authors.rb:4]: +

File: spec/factories/books.rb [hunk 8]
```diff
@@ -0,0 +1,8 @@
+FactoryBot.define do
+  factory :book do
+    id { 1 }
+    title { 'The Lord of the Rings' }
+    price { 1000 }
+    author_id { 1 }
```
[the remark this answers is not in the document]
Comment (n-chornyi) [reply spec/factories/books.rb:6]: +

File: app/models/review.rb [hunk 9]
```diff
@@ -0,0 +1,7 @@
+class Review < ApplicationRecord
+  enum status: %i[Unprocessed Approved Rejected]
```
Comment (gremax) [inline app/models/review.rb:2]: should be downcased

File: spec/models/user_spec.rb [hunk 10]
```diff
@@ -0,0 +1,7 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  it { should validate_presence_of(:email) }
+  #it { should validate_uniqueness_of(:email) }
```
Comment (gremax) [inline spec/models/user_spec.rb:5]:
```
create(:user, email: 'user@example.com')
```

**Review by gremax — APPROVED** [review]

**Pull request merged**